actors/abi/bitfield.go

```diff
@@ -0,0 +1,232 @@
+package abi
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/xerrors"
+
+	rlepluslazy "github.com/filecoin-project/specs-actors/actors/util/rleplus"
+)
+
+var ErrBitFieldTooMany = errors.New("to many items in RLE")
+
+type BitField struct {
+	rle rlepluslazy.RLE
+
+	bits map[uint64]struct{}
+}
+
+func NewBitField() BitField {
+	bf, err := NewBitFieldFromBytes([]byte{})
+	if err != nil {
+		panic(fmt.Sprintf("creating empty rle: %+v", err))
+	}
+	return bf
+}
+
+func NewBitFieldFromBytes(rle []byte) (BitField, error) {
+	bf := BitField{}
+	rlep, err := rlepluslazy.FromBuf(rle)
+	if err != nil {
+		return BitField{}, xerrors.Errorf("could not decode rle+: %w", err)
+	}
+	bf.rle = rlep
+	bf.bits = make(map[uint64]struct{})
+	return bf, nil
+
+}
+
+func BitFieldFromSet(setBits []uint64) BitField {
+	res := BitField{bits: make(map[uint64]struct{})}
+	for _, b := range setBits {
+		res.bits[b] = struct{}{}
+	}
+	return res
+}
+
+func MergeBitFields(a, b BitField) (BitField, error) {
+	ra, err := a.rle.RunIterator()
+	if err != nil {
+		return BitField{}, err
+	}
+
+	rb, err := b.rle.RunIterator()
+	if err != nil {
+		return BitField{}, err
+	}
+
+	merge, err := rlepluslazy.Sum(ra, rb)
+	if err != nil {
+		return BitField{}, err
+	}
+
+	mergebytes, err := rlepluslazy.EncodeRuns(merge, nil)
+	if err != nil {
+		return BitField{}, err
+	}
+
+	rle, err := rlepluslazy.FromBuf(mergebytes)
+	if err != nil {
+		return BitField{}, err
+	}
+
+	return BitField{
+		rle:  rle,
+		bits: make(map[uint64]struct{}),
+	}, nil
+}
+
+func (bf BitField) sum() (rlepluslazy.RunIterator, error) {
+	if len(bf.bits) == 0 {
+		return bf.rle.RunIterator()
+	}
+
+	a, err := bf.rle.RunIterator()
+	if err != nil {
+		return nil, err
+	}
+	slc := make([]uint64, 0, len(bf.bits))
+	for b := range bf.bits {
+		slc = append(slc, b)
+	}
+
+	b, err := rlepluslazy.RunsFromSlice(slc)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := rlepluslazy.Sum(a, b)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// Set ...s bit in the BitField
+func (bf BitField) Set(bit uint64) {
+	bf.bits[bit] = struct{}{}
+}
+
+// Unset removes bit from the BitField
+func (bf BitField) Unset(bit uint64) {
+	delete(bf.bits, bit)
+}
+
+// Has returns true iff bif is set in the BitField.
+func (bf BitField) Has(bit uint64) bool {
+	return bf.bits[bit] == struct{}{}
```

Comment:
Use the `_, ok := bits[bit]` style, which will stay correct if the value type changes.